pennylane/grouping/utils.py

```diff
@@ -445,7 +445,7 @@ def string_to_pauli_word(pauli_string, wire_map=None):
     # Special case: all-identity Pauli
     if pauli_string == "I" * len(wire_map):
         first_wire = list(wire_map.keys())[0]
-        return Identity(wire_map[first_wire])
+        return Identity(first_wire)
```

Comment:
Is this done to prevent having Identity for a wire that has been removed during tapering?

Reply:
Yes. Apparently, this was a tiny bug that made the program return `Identity(wires=[0])` as you can see below since it used the index of the first label (which will always be 0) instead of the wire label itself.
![image](https://user-images.githubusercontent.com/23234279/148395937-7c7d4117-7b2f-4cbd-b968-9964974f33e4.png)

Reply:
Note: This condition was actually being tested in the unit tests but was able to pass that test because of the following scenario:
```
>>> qml.Identity("a").compare(qml.Identity(1))
True
```

Reply:
You might create a bug report and fix it in a separate PR. Up to you.